src/components/generics/back-button.tsx

```diff
@@ -12,41 +11,40 @@ interface BackButtonProps {
 }
 
 /**
- * Returns the localized "Back" label for a given module key.
+ * Returns the localized "Back" label for a given module key 
+ * which also allows the translation to function properly.
  *
  * @param moduleKey - A key from the `strings` object or null
  */
 const getBackLabel = (moduleKey: keyof typeof strings | null): string => {
   if (!moduleKey) return "Back";
+
   const page = strings[moduleKey];
-  if (page && typeof page === "object" && "back" in page && typeof page.back === "string") {
-    return page.back;
-  }
-  return "Back";
+  const pageObj = typeof page === 'object' &&page !== null;
+  const incBack = pageObj && "back" in page;
+
+  return incBack ? (page as { back: string }).back : "Back";
```

Comment:
Is this a work around due to issues with the use of the locals? This:
(page as { back: string }).back

Seems very odd